client/hijack.go

```diff
@@ -205,3 +205,14 @@ type hijackedConn struct {
 func (c *hijackedConn) Read(b []byte) (int, error) {
 	return c.r.Read(b)
 }
+
+// hijackedConn must implement CloseWrite and pass it through to the underlying
+// Conn.
+var _ types.CloseWriter = &hijackedConn{}
+
+func (c *hijackedConn) CloseWrite() error {
```

Comment:
It seems we need two types and decide which one to use based on if the underlying conn supports `CloseWrite`, otherwise the caller will end up doing something like:

```go
cw, ok := hijack.(types.CloseWriter)
if ok {
    return cw.CloseWrite
}
return hijack.Close()
```

Which would lead to a leaked fd.